clients/js/src/Collection.ts

```diff
@@ -0,0 +1,382 @@
+      this.client.api.options,
+    );
+  }
+
+  /**
+   * Upsert items to the collection
+   * @param {Object} params - The parameters for the query.
+   * @param {ID | IDs} [params.ids] - IDs of the items to add.
+   * @param {Embedding | Embeddings} [params.embeddings] - Optional embeddings of the items to add.
+   * @param {Metadata | Metadatas} [params.metadatas] - Optional metadata of the items to add.
+   * @param {Document | Documents} [params.documents] - Optional documents of the items to add.
+   * @returns {Promise<void>}
+   *
+   * @example
+   * ```typescript
+   * const response = await collection.upsert({
+   *   ids: ["id1", "id2"],
+   *   embeddings: [[1, 2, 3], [4, 5, 6]],
+   *   metadatas: [{ "key": "value" }, { "key": "value" }],
+   *   documents: ["document1", "document2"],
+   * });
+   * ```
+   */
+  async upsert(params: UpsertRecordsParams): Promise<void> {
+    await this.client.init();
+
+    await this.client.api.upsert(
+      this.id,
+      // TODO: For some reason the auto generated code requires metadata to be defined here.
+      (await prepareRecordRequest(
+        params,
+        this.embeddingFunction,
+      )) as GeneratedApi.AddEmbedding,
+      this.client.api.options,
+    );
+  }
+
+  /**
+   * Count the number of items in the collection
+   * @returns {Promise<number>} - The number of items in the collection.
+   *
+   * @example
+   * ```typescript
+   * const count = await collection.count();
+   * ```
+   */
+  async count(): Promise<number> {
+    await this.client.init();
+    return (await this.client.api.count(
+      this.id,
+      this.client.api.options,
+    )) as number;
+  }
+
+  /**
+   * Get items from the collection
+   * @param {Object} params - The parameters for the query.
+   * @param {ID | IDs} [params.ids] - Optional IDs of the items to get.
+   * @param {Where} [params.where] - Optional where clause to filter items by.
+   * @param {PositiveInteger} [params.limit] - Optional limit on the number of items to get.
+   * @param {PositiveInteger} [params.offset] - Optional offset on the items to get.
+   * @param {IncludeEnum[]} [params.include] - Optional list of items to include in the response.
+   * @param {WhereDocument} [params.whereDocument] - Optional where clause to filter items by.
+   * @returns {Promise<GetResponse>} - The response from the server.
+   *
+   * @example
+   * ```typescript
+   * const response = await collection.get({
+   *   ids: ["id1", "id2"],
+   *   where: { "key": "value" },
+   *   limit: 10,
+   *   offset: 0,
+   *   include: ["embeddings", "metadatas", "documents"],
+   *   whereDocument: { $contains: "value" },
+   * });
+   * ```
+   */
+  async get({
+    ids,
+    where,
+    limit,
+    offset,
+    include,
+    whereDocument,
+  }: BaseGetParams = {}): Promise<GetResponse> {
+    await this.client.init();
+
+    const idsArray = ids ? toArray(ids) : undefined;
+
+    const resp = (await this.client.api.aGet(
```

Comment:
I think we should add validation for all the get parameters. do we have an existing ticket tracking this? If not I can create an issue

Reply:
good idea - can you make that issue?